misc/codegen/generators/qlgen.py

```diff
@@ -312,7 +312,43 @@ def _get_stub(cls: schema.Class, base_import: str, generated_import_prefix: str)
     else:
         accessors = []
     return ql.Stub(name=cls.name, base_import=base_import, import_prefix=generated_import_prefix,
-                   synth_accessors=accessors, ql_internal="ql_internal" in cls.pragmas)
+                   doc=cls.doc, synth_accessors=accessors,
+                   ql_internal="ql_internal" in cls.pragmas)
+
+
+def _patch_class_qldocs(cls: str, qldoc: str, stub_file: pathlib.Path):
+    if not qldoc or not stub_file.exists():
+        return
+    qldoc = "\n".join(l.rstrip() for l in qldoc.splitlines())
+    tmp = stub_file.with_suffix(f'{stub_file.suffix}.bkp')
+    header = "// the following QLdoc is generated: if you need to edit it, do it in the schema file\n"
+    with open(stub_file) as input:
+        qldoc_start = None
+        qldoc_end = None
+        class_start = None
+        for lineno, line in enumerate(input, 1):
+            if line == header:
+                qldoc_start = lineno
+            if line.startswith("/**") and lineno - 1 != qldoc_start:
```

Comment:
Does this / should this work if the QLDoc is indented?

Reply:
no, it does not, but we only target the main class QLdoc here, which won't ever be indented after QL formatting. If this changes at some point in the future (which I doubt), we would get the assertions fired later and we would need to fix the logic.